spec/javascripts/exercise-spec.js

```diff
@@ -3,12 +3,28 @@ describe('exercise', () => {
   it('current exercise information is available when present', () => {
     $('body').html(`
     <input type="hidden" name="mu-exercise-id" id="mu-exercise-id" value="3361" />
-    <input type="hidden" name="mu-exercise-layout" id="mu-exercise-layout" value="input_right" />`)
+    <input type="hidden" name="mu-exercise-layout" id="mu-exercise-layout" value="input_right" />
+    <input type="hidden" name="mu-exercise-settings" id="mu-exercise-settings" value="{}" />`)
```

Comment:
This won't 'scale up' much. We will need to replace this by an exercise resource-hash in the future if more exercise attributes are needed.

Reply:
PS: in case you are wondering, `settings`  can never be `nil`.